File: telegram/src/main/java/by/mrrockka/bot/TelegramBotsProperties.java
package by.mrrockka.bot;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties(prefix = "telegrambots")
@Data
public class TelegramBotsProperties {

  private String name;
  private String nickname;
  private boolean test;
  private String token;
}

File: telegram/src/main/java/by/mrrockka/bot/TelegramBotsProperties.kt
package by.mrrockka.bot

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "telegrambots")
class TelegramBotsProperties {
    lateinit var name: String
    lateinit var nickname: String
    lateinit var token: String
    var enabled: Boolean = false
    val botpath: String by lazy { "/bot$token" }
}
